src/main/cpp/src/format_float.cu

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cast_string.hpp"
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/iterator.cuh>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/strings/detail/convert/int_to_string.cuh>
+#include <cudf/strings/detail/converters.hpp>
+#include <cudf/strings/detail/strings_children.cuh>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/utilities/default_stream.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <ftos_converter.cu>
+
+using namespace cudf;
+
+namespace spark_rapids_jni {
+
+namespace detail {
+namespace {
+
+template <typename FloatType>
+struct format_float_fn {
+  column_device_view d_floats;
+  int digits;
+  size_type* d_offsets;
+  char* d_chars;
+
+  __device__ size_type compute_output_size(FloatType value, int digits)
+  {
+    ftos_converter fts;
+    bool is_float = std::is_same_v<FloatType, float>;
+    return static_cast<size_type>(fts.compute_format_float_size(static_cast<double>(value), digits, is_float));
+  }
+
+  __device__ void format_float(size_type idx, int digits)
+  {
+    FloatType value = d_floats.element<FloatType>(idx);
+    ftos_converter fts;
+    bool is_float = std::is_same_v<FloatType, float>;
+    fts.format_float(static_cast<double>(value), digits, d_chars + d_offsets[idx], is_float);
```

Comment:
Calling this function is also confusing as we don't know where is the output.
```suggestion
    auto const output = d_chars + d_offsets[idx];
    fts.format_float(static_cast<double>(value), digits, is_float, output);
```
(output should be at the end of the param list)